lib/serverless/api-gateway.js

```diff
@@ -158,7 +158,7 @@ const requiredHttpApiV1Keys = [
 
 // See https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html
 function isGatewayV1Event(event) {
-  if (event?.requestContext?.elb !== undefined) {
+  if (event?.requestContext?.elb !== undefined || event?.version === undefined) {
```

Comment:
Do we not need the same check on line 176?

Reply:
hmm we can do it also here for the sake of it, I works atm because in L179 we check the keys and version 2.0 so even if the check would return true, as long as the custom payload has not by accident also version with 2.0 it will fail